src/main/java/org/wso2/carbon/connector/connection/EmailConnection.java

```diff
@@ -49,7 +59,26 @@ public EmailConnection(ConnectionConfiguration connectionConfiguration) {
             sessionProperties.putAll(setSecureProperties(connectionConfiguration));
         }
 
+        if (connectionConfiguration.isOAuth2Enabled()) {
+            sessionProperties.setProperty(protocol.getAuthMechanismsProperty(), EmailConstants.AUTH_MECHANISM_XOAUTH2);
+        }
+
         if (connectionConfiguration.getRequireAuthentication()) {
+            if (connectionConfiguration.isOAuth2Enabled()) {
+                log.debug("Attempting to connect to " + connectionConfiguration.getProtocol().getName() +
```

Comment:
Is there an option to check whether debug logs are enabled here?

Reply:
Fixed.